Lib/functools.py

```diff
@@ -690,7 +674,6 @@ def wrapper(*args, **kwds):
                     # still adjusting the links.
                     root = oldroot[NEXT]
                     oldkey = root[KEY]
-                    oldresult = root[RESULT]
```

Comment:
This value was never used

Reply:
This line needs to be restored.  Its job is to hold on to the object reference until the function returns (after the cache is in a stable state).

Reply:
Good catch! Sorry about that, I wasn't aware. My IDE complained about it and I should've doubted it more. I reverted 5ebba28c82d470ce0adddbee6185415065c953bd